File: jcr_root/apps/training/components/structure/site-topnav/topnav.js
use(function () {
   var items = [];
    var root = currentPage.getAbsoluteParent(2);
    if(root == null){
        root = currentPage;
    }
    
    log.info("############### Root page is : " + root.getTitle());
    
    var it = root.listChildren(new Packages.com.day.cq.wcm.api.PageFilter());
    while(it.hasNext()){
        var page = it.next();
        items.push(page);
    }
    
    return {
        items: items,
        root: root
    }
});

File: jcr_root/apps/training/components/structure/site-topnav/TopNav.java
package apps.training.components.structure.site_topnav;

import java.util.*;
import java.util.Iterator;
import com.day.cq.wcm.api.Page;
import com.day.cq.wcm.api.PageFilter;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import com.adobe.cq.sightly.WCMUsePojo;

public class TopNav extends WCMUsePojo{
    private List<Page> items = new ArrayList<Page>();
    private Page rootPage;
    private static final Logger LOG = LoggerFactory.getLogger(TopNav.class);

    // Initializes the navigation
    @Override
    public void activate() throws Exception {
        rootPage = getCurrentPage().getAbsoluteParent(2);

        if (rootPage == null) {
        	rootPage = getCurrentPage();
        }
        
        LOG.info("##### Root page is : " + rootPage.getTitle());
        
        Iterator<Page> childPages = rootPage.listChildren(new PageFilter(getRequest()));
	   	while (childPages.hasNext()) {
			items.add(childPages.next());
	   	}
    }

    // Returns the navigation items
    public List<Page> getItems() {
        return items;
    }
    // Returns the navigation root
    public Page getRoot() {
        return rootPage;
    }
}